src/Explorer/Menus/CommandBar/CommandBarComponentButtonFactory.tsx

```diff
@@ -134,13 +134,19 @@ export function createContextCommandBarButtons(container: Explorer): CommandButt
   const buttons: CommandButtonComponentProps[] = [];
 
   if (!container.isDatabaseNodeOrNoneSelected() && userContext.apiType === "Mongo") {
-    const label = "New Shell";
+    const showMongoTerminal =
+      container.isNotebookEnabled() && !userContext.databaseAccount.properties.isVirtualNetworkFilterEnabled;
```

Comment:
>isVirtualNetworkFilterEnabled [](http://example.com/codeflow?start=79&length=29)

may be we need addnal property check for iprules

Reply:
I checked tha iprules array is populated only when isVirtualNetworkFilterEnabled is true. Any particular reason to add ipRUles check as well?

Reply:
Is there a possibility that the account has VNET disabled but has firewall enabled? If yes we may need additional check here.